pkg/users/account-create-dialog.js

```diff
@@ -0,0 +1,284 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+
+import { Validated } from "./dialog-utils.js";
+import { password_quality, passwd_change } from "./password-dialogs.js";
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+function AccountCreateBody({ state, errors, change }) {
+    const {
+        real_name, user_name,
+        password, password_confirm, password_strength, password_message,
+        locked
+    } = state;
```

Comment:
> I don't like this kind of "nested" props dict here, it's a very unusual pattern. It especially shouldn't be called "state", as this is a presentational components that only has properties.

It's the dialog state.  I don't think it is unusual that the state of the parent component is used as props for the children, and why not call it "state"?

Reply:
I have pushed a FIXUP that arranges things slightly differently.  Now the dialog state is 'hidden' in a context and we have things like `DialogTextInput` components that know how to work with such a context.

But I guess you want to make the state more visible instead of less, right?

Reply:
Right, now the API is totally unclear. As I said, I don't want to block this PR on such bikeshedding, but this makes components much harder to understand and also more brittle as there is zero type checking.